src/Mantasflowers.Services/DataAccess/Repositories/OrderRepository.cs

```diff
@@ -24,5 +24,18 @@ public async Task<Order> GetDetailedOrderAsync(Guid id)
 
             return order;
         }
+
+        public async override Task<Order> CreateAsync(Order entity)
+        {
+            foreach (var item in entity.OrderItems)
+            {
+                item.Product = await _dbContext.Products.FindAsync(item.ProductId);
+                _dbContext.Entry(item.Product).State = EntityState.Unchanged;
```

Comment:
Does this have something to do about what we talked on Monday? Looks hacky.

Reply:
It's rather not hacky, but the proper solution, when adding new entities with relation to existing ones. It also prevents Order from creating (inserting) new products, or, at the very least, from trying to.